admin/services/check-window.service.js

```diff
@@ -114,6 +93,88 @@ const checkWindowService = {
     const activeCheckWindow = await checkWindowDataService.sqlFindOneActiveCheckWindow(currentCheck.checkWindow_id)
     if (!activeCheckWindow) throw new Error('There is no open check window')
     return activeCheckWindow
+  },
+
+  /**
+   * Get all check windows
+   * @param {String} sortField
+   * @param {String} sortDirection
+   * @returns {Object}
+   */
+  getAllCheckWindows: async(sortField, sortDirection) => {
+    // fetch past current and future check windows
+    const checkWindows = await checkWindowDataService.sqlFindAllCheckWindows(sortField, sortDirection)
+    const pastCheckWindows = []
+    const newCheckWindows = []
+    checkWindows.map(cw => {
+      const adminStartDate = moment(cw.adminStartDate)
+      const checkStartDate = moment(cw.checkStartDate)
+      const checkEndDate = moment(cw.checkEndDate)
+
+      if (moment(checkStartDate).isAfter(moment(checkEndDate))) {
+        throw new Error('Check start date is after check end date')
+      }
+
+      const isPast = moment(checkEndDate).isBefore(moment.now())
+      const isCurrent = moment(checkStartDate).isSameOrBefore(moment.now()) && moment(checkEndDate).isSameOrAfter(moment.now())
+      const isFuture = moment(checkStartDate).isAfter(moment.now())
+
+      const checkWindow = {
+        id: cw.id,
+        name: cw.name,
+        adminStartDate: adminStartDate.format('D MMM YYYY'),
+        checkDates: dateService.formatCheckPeriod(checkStartDate, checkEndDate),
+        isPast
+      }
+
+      if (isPast) {
+        checkWindow.canRemove = false
+        pastCheckWindows.push(checkWindow)
```

Comment:
Adding `checkWindow` to different arrays and then concatenting the arrays doesn't actually seem to be required.  The code would be very slightly simpler without that step.  Is there a reason for it?